File: solutions/go/hamming/1/hamming.go
// package hamming provides functions to calculate the hamming distance
package hamming

import "errors"

// Distance calculates the hamming distance between two strings
func Distance(a, b string) (int, error) {
	if len(a) != len(b) {
		return -1, errors.New("lengths must be the same")
	}

	var count int
	for i := range a {
		if a[i] != b[i] {
			count++
		}
	}
	return count, nil
}

File: solutions/cpp/hamming/1/hamming.cpp
#include "hamming.h"

namespace hamming {

    int compute(const std::string &first_string, const std::string &second_string) {
        unsigned int str_len = first_string.length();

        if (str_len != second_string.length()) {
            throw std::domain_error("String lengths must be equal");
        }

        int count = 0;
        for (int i = 0; i < str_len; i++) {
            if (first_string[i] != second_string[i]) {
                count++;
            }
        }
        return count;
    }
}  // namespace hamming
